### PR TITLE
feat(databricks-driver): add db catalog

### PR DESCRIPTION
**Pull request by frolovdev**
**Check List**
- [ ] Tests has been run in packages where changes made if available
- [ ] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

[For example #12]

**Description of Changes Made (if issue reference is not provided)**

[Description goes here]

### CODE REVIEW

File: packages/cubejs-query-orchestrator/src/orchestrator/PreAggregations.ts [hunk 1]
```diff
@@ -1998,11 +1998,14 @@ export class PreAggregations {
...
+
+    const dbCatalogStringPart = dbCatalog ? `${dbCatalog}.` : '';
     if (versionEntry.naming_version === 2) {
-      return `${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${encodeTimeStamp(versionEntry.last_updated_at)}`;
+      return `${dbCatalogStringPart}${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${encodeTimeStamp(versionEntry.last_updated_at)}`;
     }
 
-    return `${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${versionEntry.last_updated_at}`;
+    return `${dbCatalogStringPart}${versionEntry.table_name}_${versionEntry.content_version}_${versionEntry.structure_version}_${versionEntry.last_updated_at}`;
   }
```
Comment (frolovdev) [inline packages/cubejs-query-orchestrator/src/orchestrator/PreAggregations.ts:2009]: 📓 required for external: false and temp table strategy, in any other case we will create table in hive_metastore by default

File: packages/cubejs-databricks-jdbc-driver/src/DatabricksDriver.ts [hunk 2]
```diff
@@ -217,16 +221,25 @@ export class DatabricksDriver extends JDBCDriver {
     }));
   }
 
+  private getFullDbName(dbName?: string) {
```
Comment (buntarb) [inline packages/cubejs-databricks-jdbc-driver/src/DatabricksDriver.ts:224]: Why do you need this as a separate method?

File: packages/cubejs-query-orchestrator/src/orchestrator/PreAggregations.ts [hunk 3]
```diff
@@ -1998,11 +1998,14 @@ export class PreAggregations {
...
 
   public static targetTableName(versionEntry): string {
+    const dbCatalog = getEnv('dbCatalog');
+
+    const dbCatalogStringPart = dbCatalog ? `${dbCatalog}.` : '';
```
Comment (buntarb) [inline packages/cubejs-query-orchestrator/src/orchestrator/PreAggregations.ts:2003]: You need to use the `versionEntry`, meaning the `dbCatalog` value should be used at the initial moment and not here.

**Review by buntarb — CHANGES_REQUESTED** [review]

File: packages/cubejs-databricks-jdbc-driver/src/DatabricksDriver.ts (hunk 2 above)
Comment (frolovdev) [reply packages/cubejs-databricks-jdbc-driver/src/DatabricksDriver.ts:224]: used in two places, looks like a reasonable thing

**Pull request closed without merging**